examples/real-dom.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var document = require('global/document');
+var hg = require('../index.js');
+var h = require('../index.js').h;
+var setTimeout = require('timers').setTimeout;
+
+function Hook() {}
+// Hook to be called when DOM node has been rendered
+Hook.prototype.hook = function hook(node) {
+    node.innerHTML = 'Set directly on real DOM node, by <em>hook</em>.';
+};
+// Hook to be called when DOM node is removed
+Hook.prototype.unhook = function unhook(node) {
+    node.innerHTML = 'Set directly on real DOM node, by <em>unhook</em>.';
+};
+
+function Widget() {
+    this.type = 'Widget';
+}
+
+Widget.prototype.init = function init() {
+    var elem = document.createElement('div');
+    elem.innerHTML = 'Set directly on real DOM node, by widget ' +
```

Comment:
innerHTML in widgets is fine.

innerHTML in hooks breaks the tree indexing that virtual-dom is using.

In a hook; virtual dom owns the dom element and its children. You should not mutate the DOM tree in a hook; you should only mutate attributes on a DOM node.